From 2d580928a789446d86b1a510569745c37b0172d3 Mon Sep 17 00:00:00 2001
From: root <root@congroo>
Date: Fri, 14 Mar 2025 13:10:48 +0800
Subject: [PATCH] Applied adapted patch for linux-6.12.y

---
 drivers/net/ethernet/ibm/ibmvnic.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ibm/ibmvnic.c b/drivers/net/ethernet/ibm/ibmvnic.c
index 97425c06e1ed..4260cdcd308b 100644
--- a/drivers/net/ethernet/ibm/ibmvnic.c
+++ b/drivers/net/ethernet/ibm/ibmvnic.c
@@ -2402,6 +2402,7 @@ static netdev_tx_t ibmvnic_xmit(struct sk_buff *skb, struct net_device *netdev)
 	unsigned int tx_map_failed = 0;
 	union sub_crq indir_arr[16];
 	unsigned int tx_dropped = 0;
+	unsigned int skblen;
 	unsigned int tx_packets = 0;
 	unsigned int tx_bytes = 0;
 	dma_addr_t data_dma_addr;
@@ -2610,8 +2611,9 @@ static netdev_tx_t ibmvnic_xmit(struct sk_buff *skb, struct net_device *netdev)
 		netif_stop_subqueue(netdev, queue_num);
 	}
 
+	skblen = skb->len;
 	tx_packets++;
-	tx_bytes += skb->len;
+	tx_bytes += skblen;
 	txq_trans_cond_update(txq);
 	ret = NETDEV_TX_OK;
 	goto out;
-- 
2.25.1

