From 8a1475b211f1d38bb1c144a14053a908f8811897 Mon Sep 17 00:00:00 2001
From: root <root@congroo0>
Date: Thu, 20 Mar 2025 07:40:02 +0800
Subject: [PATCH] Applied adapted patch for linux-6.12.y

---
 drivers/net/geneve.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/net/geneve.c b/drivers/net/geneve.c
index ba15a0a4ce62..f810b5d8427e 100644
--- a/drivers/net/geneve.c
+++ b/drivers/net/geneve.c
@@ -1902,14 +1902,7 @@ static void geneve_destroy_tunnels(struct net *net, struct list_head *head)
 {
 	struct geneve_net *gn = net_generic(net, geneve_net_id);
 	struct geneve_dev *geneve, *next;
-	struct net_device *dev, *aux;
 
-	/* gather any geneve devices that were moved into this ns */
-	for_each_netdev_safe(net, dev, aux)
-		if (dev->rtnl_link_ops == &geneve_link_ops)
-			unregister_netdevice_queue(dev, head);
-
-	/* now gather any other geneve devices that were created in this ns */
 	list_for_each_entry_safe(geneve, next, &gn->geneve_list, next) {
 		/* If geneve->dev is in the same netns, it was already added
 		 * to the list by the previous loop.
-- 
2.25.1

