From 71cc324fb1f03a9f8f18357ee5d15e55a9dd4e86 Mon Sep 17 00:00:00 2001
From: root <root@congroo>
Date: Thu, 26 Dec 2024 18:19:29 +0800
Subject: [PATCH] Adapted patch  for linux-6.12.y

---
 arch/arm64/kvm/sys_regs.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm64/kvm/sys_regs.c b/arch/arm64/kvm/sys_regs.c
index fbed433283c9..286a5aa9ebae 100644
--- a/arch/arm64/kvm/sys_regs.c
+++ b/arch/arm64/kvm/sys_regs.c
@@ -1812,6 +1812,13 @@ static u64 read_sanitised_id_dfr0_el1(struct kvm_vcpu *vcpu,
 
 	val = ID_REG_LIMIT_FIELD_ENUM(val, ID_DFR0_EL1, CopDbg, Debugv8p8);
 
+	/*
+	 * MPAM is disabled by default as KVM also needs a set of PARTID to
+	 * program the MPAMVPMx_EL2 PARTID remapping registers with. But some
+	 * older kernels let the guest see the ID bit.
+	 */
+	val &= ~ID_AA64PFR0_EL1_MPAM_MASK;
+
 	return val;
 }
 
-- 
2.25.1

