From 4659601eb1f8ea20ca9758b09a884288df55defd Mon Sep 17 00:00:00 2001
From: root <root@congroo>
Date: Thu, 23 Jan 2025 18:15:47 +0800
Subject: [PATCH] Adapted patch for linux-6.12.y

---
 include/linux/static_call.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/linux/static_call.h b/include/linux/static_call.h
index 78a77a4ae0ea..59700baca14a 100644
--- a/include/linux/static_call.h
+++ b/include/linux/static_call.h
@@ -162,6 +162,8 @@ extern void arch_static_call_transform(void *site, void *tramp, void *func, bool
 
 extern int static_call_initialized;
 
+extern int static_call_initialized;
+
 extern int __init static_call_init(void);
 
 extern void static_call_force_reinit(void);
@@ -229,6 +231,8 @@ extern long __static_call_return0(void);
 
 #define static_call_initialized 0
 
+#define static_call_initialized 0
+
 static inline int static_call_init(void) { return 0; }
 
 #define DEFINE_STATIC_CALL(name, _func)					\
@@ -287,6 +291,8 @@ extern long __static_call_return0(void);
 
 #define static_call_initialized 0
 
+#define static_call_initialized 0
+
 static inline int static_call_init(void) { return 0; }
 
 static inline long __static_call_return0(void)
-- 
2.25.1

