From c8a87d22036e62bac0c6f7836078d8103caa6457 Mon Sep 17 00:00:00 2001
From: Thomas Loimer <thomas.loimer@tuwien.ac.at>
Date: Wed, 22 Jan 2025 23:27:43 +0100
Subject: [PATCH] Reject arcs with co-incident points, ticket #186


diff --git a/fig2dev/tests/read.at b/fig2dev/tests/read.at
index 1b4baea..da9ea3e 100644
--- a/fig2dev/tests/read.at
+++ b/fig2dev/tests/read.at
@@ -14,7 +14,7 @@ dnl party to do so, with the only requirement being that the above copyright
 dnl and this permission notice remain intact.
 
 dnl read.at
-dnl Author: Thomas Loimer, 2017-2024
+dnl Author: Thomas Loimer, 2017-2025
 
 
 AT_BANNER([Sanitize and harden input.])
