From c2f265c8ed2c5a52ebb775bf987808c4f43bfda2 Mon Sep 17 00:00:00 2001
From: root <root@congroo>
Date: Thu, 16 Jan 2025 17:37:32 +0800
Subject: [PATCH] Adapted patch for linux-6.12.y

---
 io_uring/io_uring.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/io_uring/io_uring.c b/io_uring/io_uring.c
index b2736e3491b8..bd6f53c30ddc 100644
--- a/io_uring/io_uring.c
+++ b/io_uring/io_uring.c
@@ -515,7 +515,11 @@ static void io_queue_iowq(struct io_kiocb *req)
 	struct io_uring_task *tctx = req->task->io_uring;
 
 	BUG_ON(!tctx);
-	BUG_ON(!tctx->io_wq);
+
+	if ((current->flags & PF_KTHREAD) || !tctx->io_wq) {
+		io_req_task_queue_fail(req, -ECANCELED);
+		return;
+	}
 
 	/* init ->work of the whole link before punting */
 	io_prep_async_link(req);
-- 
2.25.1

