From 88b619acb9494fa2ad281ba4202046d9fcffa2c7 Mon Sep 17 00:00:00 2001
From: root <root@congroo>
Date: Thu, 26 Dec 2024 18:33:13 +0800
Subject: [PATCH] Adapted patch  for linux-6.12.y

---
 rust/Makefile | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/rust/Makefile b/rust/Makefile
index 9f59baacaf77..e259449f8cc5 100644
--- a/rust/Makefile
+++ b/rust/Makefile
@@ -58,6 +58,11 @@ alloc-cfgs = \
     --cfg no_rc \
     --cfg no_sync
 
+alloc-cfgs = \
+    --cfg no_global_oom_handling \
+    --cfg no_rc \
+    --cfg no_sync
+
 quiet_cmd_rustdoc = RUSTDOC $(if $(rustdoc_host),H, ) $<
       cmd_rustdoc = \
 	OBJTREE=$(abspath $(objtree)) \
@@ -327,6 +332,9 @@ $(obj)/exports_core_generated.h: $(obj)/core.o FORCE
 $(obj)/exports_alloc_generated.h: $(obj)/alloc.o FORCE
 	$(call if_changed,exports)
 
+$(obj)/exports_alloc_generated.h: $(obj)/alloc.o FORCE
+	$(call if_changed,exports)
+
 # Even though Rust kernel modules should never use the bindings directly,
 # symbols from the `bindings` crate and the C helpers need to be exported
 # because Rust generics and inlined functions may not get their code generated
-- 
2.25.1

